### PR TITLE
Explode slice cli

### PR DESCRIPTION
**Pull request by benjeffery**
Add CLI commands for explode init/slice/finalise.
Also writes number of partitions to disk as discussed in #71 

Will now test this on GeL VCFs

### CODE REVIEW

**Comment by benjeffery** [comment]
Exploded chr20 of GeL in 1hr, but now investigating why about 10% of the jobs don't write the partition, but also do not error.

**Comment by benjeffery** [comment]
Failures were due to cluster issues, not this PR!

File: bio2zarr/vcf.py [hunk 1]
```diff
@@ -141,6 +148,7 @@ class VcfMetadata:
     fields: list
     partitions: list = None
     contig_lengths: list = None
+    finalised: bool = False
```
Comment (jeromekelleher) [inline bio2zarr/vcf.py:151]: As discussed, let's use a different file to flag partially exploded

File: bio2zarr/vcf.py [hunk 2]
```diff
@@ -940,70 +981,108 @@ def convert_partition(
                     # is that we get a significant pause at the end of the counter while
                     # all the "small" fields get flushed. Possibly not much to be done about it.
                     core.update_progress(1)
-
+                partition_metadata = {
+                    "num_records": num_records,
+                    "field_summaries": {k:v.asdict() for k,v in tcw.field_summaries.items()}
+                }
+                with open(out_path / f"partition_{partition_index}_metadata.json", "w") as f:
```
Comment (jeromekelleher) [inline bio2zarr/vcf.py:988]: We've used ``p{j}`` as the partition file name elsewhere, may as well stick with this, ``e.g. p0_summary.json``

File: bio2zarr/vcf.py [hunk 3]
```diff
@@ -940,70 +981,108 @@ def convert_partition(
...
-                    out_path,
-                    column_chunk_size=column_chunk_size,
-                )
-            num_records = 0
-            partition_summaries = []
-            for index, summary, num_records in pwm.results_as_completed():
-                partition_summaries.append(summary)
-                vcf_metadata.partitions[index].num_records = num_records
-
+        if stop-start == 1:
```
Comment (jeromekelleher) [inline bio2zarr/vcf.py:1036]: I'd rather just use the same code path, if only for progress tracking reasons. It's not worth the complexity to keep this case a bit quicker.

**Review by jeromekelleher — COMMENTED** [review]
Basically looks good, a few minor details need fixing up. I can do the refactoring into PickleChunkedVcfWriter later.

**Comment by benjeffery** [comment]
Comments addressed.
Also found the "zeros in summary" error and added more testing. 
I'll squash down before merge when it looks ready.

**Comment by jeromekelleher** [comment]
Great - I'll merge and update. Going to try and get breaking changes to formats done ASAP

**Pull request merged**